### PR TITLE
Remove TableDiff::$name and getName()

### PR DESCRIPTION
**Pull request by morozov**
The API being removed was deprecated in https://github.com/doctrine/dbal/pull/5678.

Additionally, this patch makes the reference from the table diff to the old table mandatory. In order to do that, it moves the corresponding constructor argument from the last position to the first, since all remaining arguments are still optional. This change should not affect the consumers of the DBAL API since the constructor is internal.

### CODE REVIEW

File: tests/Platforms/SQLServerPlatformTestCase.php [hunk 1]
```diff
@@ -1224,7 +1228,7 @@ public static function getGeneratesIdentifierNamesInAlterTableSQL(): iterable
             // Quoted identifiers non-reserved keywords.
             [
                 new TableDiff(
-                    '`mytable`',
+                    new Table('mytable'),
```
Comment (SenseException) [inline tests/Platforms/SQLServerPlatformTestCase.php]: Is this supposed to keep the `?
```suggestion
                    new Table('`mytable`'),
```

File: tests/Platforms/SQLServerPlatformTestCase.php [hunk 2]
```diff
@@ -1300,7 +1304,7 @@ public static function getGeneratesIdentifierNamesInAlterTableSQL(): iterable
             // Quoted identifiers reserved keywords.
             [
                 new TableDiff(
-                    '`table`',
+                    new Table('mytable'),
```
Comment (SenseException) [inline tests/Platforms/SQLServerPlatformTestCase.php]: Is this supposed to keep the `?
```suggestion
                    new Table('`mytable`'),
```

File: tests/Platforms/SQLServerPlatformTestCase.php (hunk 1 above)
Comment (morozov) [reply tests/Platforms/SQLServerPlatformTestCase.php]: Thanks! The quotes got lost during refactoring.

File: tests/Platforms/SQLServerPlatformTestCase.php (hunk 2 above)
Comment (morozov) [reply tests/Platforms/SQLServerPlatformTestCase.php]: The table name got accidentally changed during the refactoring but apparently, it's irrelevant to the test.

**Review by greg0ire — APPROVED** [review]

**Pull request merged**